### PR TITLE
docs: typos in DEVELOPMENT.md and README.md

### PR DESCRIPTION
**Pull request by PoulavBhowmick03**
# Description

While going through the project and the documentation, I came across some typos and thought this could just be some issues to get started with in solving, as I explore the entire codebase more.
## Tests

- [] Yes
- [X] No, because they aren't needed
- [ ] No, because I need help

## Added to documentation?

- [ ] README.md
- [ ] [Dojo Book](https://github.com/dojoengine/book)
- [X] No documentation needed

## Checklist

- [ ] I've formatted my code (`scripts/prettier.sh`, `scripts/rust_fmt.sh`, `scripts/cairo_fmt.sh`)
- [ ] I've linted my code (`scripts/clippy.sh`, `scripts/docs.sh`)
- [ ] I've commented my code
- [X] I've requested a review after addressing the comments

## Summary by CodeRabbit

- **Documentation**
  - Updated links and corrected typos in `DEVELOPMENT.md`.
  - Modified description of the Dojo framework in `README.md` from "permissionless" to "permission-less".

- **Style**
  - Standardized path delimiters and quote styles in multiple TOML configuration files for consistency across `abi` field assignments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 [tg-badge]: https://img.shields.io/endpoint?color=neon&logo=telegram&label=chat&style=flat-square&url=https%3A%2F%2Ftg.sumanjay.workers.dev%2Fdojoengine
 [tg-url]: https://t.me/dojoengine
 
-Dojo is a developer friendly framework for building **provable** Games, Autonomous Worlds and other Applications that are natively composable, extensible, permissionless and persistent. It is an extension of [Cairo](https://www.cairo-lang.org/), an efficiently provable language, that supports generation of zero-knowledge proofs attesting to a computations validity and enables exponential scaling of onchain computation while maintaining the security properties of Ethereum.
+Dojo is a developer friendly framework for building **provable** Games, Autonomous Worlds and other Applications that are natively composable, extensible, permission-less and persistent. It is an extension of [Cairo](https://www.cairo-lang.org/), an efficiently provable language, that supports generation of zero-knowledge proofs attesting to a computations validity and enables exponential scaling of onchain computation while maintaining the security properties of Ethereum.
```
Comment (kariy) [inline README.md:9]: i dont have a phd in the english language but i think 'permissionless' is still considered correct.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ This guide outlines the steps to set up a development environment for Dojo. If y
 
 - [Rust](https://github.com/rust-lang/rust)
 - [Cairo](https://github.com/starkware-libs/cairo)
-- [protoc](https://github.com/protocolbuffers/protobuf)
+- [Protobuf](https://github.com/protocolbuffers/protobuf)
```
Comment (kariy) [inline DEVELOPMENT.md:9]: i think `protoc` is more accurate here though as we need user to have the `protoc` binary but i understand that it's not consistent with the other two. so we're not gonna accept this correction.

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ Install the [Cairo 1.0](https://marketplace.visualstudio.com/items?itemName=star
 
 Before you submit your pull request, you should run the test suite locally to make sure your changes haven't broken anything.
 
-To run the test, you can execute the same command that will be exected on the CI by checking the [`.github/workflows/ci.yml`](.github/workflows/ci.yml) file.
+To run the test, you can execute the same command that will be executed on the CI by checking the [`.github/workflows/ci.yml`](.github/workflows/ci.yml) file.
```
Comment (kariy) [inline DEVELOPMENT.md:41]: now this is an actual typo. good catch!

**Review by kariy — CHANGES_REQUESTED** [review]

**Comment by PoulavBhowmick03** [comment]
@kariy can you check once
Also, can you kindly suggest some resources where I can see some onchain games built with dojo? would be extremely helpful, was going through dojo docs and would love to have some ideas on which me and my team could build

**Pull request closed without merging**